editor/components/block-settings-menu/block-remove-button.js

```diff
@@ -48,7 +48,11 @@ export default compose(
 			isLocked: some( castArray( clientIds ), ( clientId ) => {
 				const rootClientId = getBlockRootClientId( clientId );
 				const templateLock = getTemplateLock( rootClientId );
-				return templateLock === 'all';
+
+				// Prevents deletion if templateLock is "all" or "insert"
+				// (if a user can't insert blocks they shouldn't be able
+				// to remove them either)
+				return templateLock === 'all' || templateLock === 'insert';
```

Comment:
Minor: In other places we do `!! templateLock` (inserting blocks).

I think we should add a similar check to `SharedBlockDeleteButton`? what do you think? in case of a template containing a shared block.

Reply:
Makes sense to me, on both counts. But I didn't know a template could include a shared block — how does that work?

Reply:
(Like, how do you know the name of the block slug to put into the template? Is it just the name you save, or a kebab-cased version of that?)

Reply:
Try adding a shared block to a regular post and copy the result (it should be `core/block` with a `ref` attribute)